pyproject.toml

```diff
@@ -23,14 +23,19 @@ dependencies = [
     "numpy==1.21.4",
     "pandas==1.3.5",
     "ujson==5.3.0",
+    "stream-unzip==0.0.70",
```

Comment:
I included this in a comment below as well (in code), but `zipfile.ZipFile` doesn't work on non-seekable streams, which means that it won't work "natively" on S3/HTTP streams without first copying the file to disk (or, wrapping it and buffering at least part of the stream in memory). `stream-unzip` instead does exactly what the name would imply - it can take as input some stream, and unzip that stream on-the-fly, yielding chunks of the underlying file as we iterate over the stream